### ISSUE TITLE
promql.parseBrokenJson panic: runtime error: slice bounds out of range

### ISSUE THREAD

**Issue by jeanralphaviles** [prometheus/prometheus#6070]
## Bug Report
**What did you do?**
Ran any Prometheus docker image from Dockerhub past **v2.12.0-rc.0** pointed at my existing data.

**What did you expect to see?**
Prometheus starting normally.

**What did you see instead? Under which circumstances?**

Prometheus crashes on startup with 

```
prometheus_1            | level=warn ts=2019-09-28T16:52:31.306Z caller=main.go:282 deprecation_notice="'storage.tsdb.retention' flag is deprecated use 'storage.tsdb.retention.time' instead."                                                                                            
prometheus_1            | level=info ts=2019-09-28T16:52:31.306Z caller=main.go:329 msg="Starting Prometheus" version="(version=2.12.0-rc.0, branch=HEAD, revision=eaf81808d5aa19ee4b9b8980ac5feeaec51408ac)"                                                                              
prometheus_1            | level=info ts=2019-09-28T16:52:31.306Z caller=main.go:330 build_context="(go=go1.12.8, user=root@e8effea18820, date=20190814-17:42:12)"                                                                                                                          
prometheus_1            | level=info ts=2019-09-28T16:52:31.306Z caller=main.go:331 host_details="(Linux 5.0.0-25-generic #26-Ubuntu SMP Thu Aug 1 12:04:58 UTC 2019 x86_64 4eb21f7bcef1 (none))"                                                                                          
prometheus_1            | level=info ts=2019-09-28T16:52:31.306Z caller=main.go:332 fd_limits="(soft=1048576, hard=1048576)"
prometheus_1            | level=info ts=2019-09-28T16:52:31.306Z caller=main.go:333 vm_limits="(soft=unlimited, hard=unlimited)"
prometheus_1            | panic: runtime error: slice bounds out of range
prometheus_1            |
prometheus_1            | goroutine 1 [running]:
prometheus_1            | github.com/prometheus/prometheus/promql.parseBrokenJson(0xc000906000, 0x4e21, 0x4e21, 0x2a6b7c0, 0xc0005dc240, 0x0, 0x0, 0xc0006281e0)                                                                                                                           
prometheus_1            |       /app/promql/query_logger.go:45 +0x131
prometheus_1            | github.com/prometheus/prometheus/promql.logUnfinishedQueries(0xc0006281e0, 0x1f, 0x4e21, 0x2a6b7c0, 0xc0005dc240)
prometheus_1            |       /app/promql/query_logger.go:70 +0x552
prometheus_1            | github.com/prometheus/prometheus/promql.NewActiveQueryTracker(0x7ffe73792ed0, 0x10, 0x14, 0x2a6b7c0, 0xc0005dc240, 0x2a6b7c0)                                                                                                                                    
prometheus_1            |       /app/promql/query_logger.go:108 +0x131
prometheus_1            | main.main()
prometheus_1            |       /app/cmd/prometheus/main.go:361 +0x52bd
```

Any version before **v2.12.0-rc.0** runs as expected.

**Environment**

* System information:

Running `prom/prometheus:v2.12.0-rc.0`

```
$ uname -srm
Linux 5.0.0-25-generic x86_64
```

* Prometheus version:

```
$ prometheus --version
prometheus, version 2.12.0-rc.0 (branch: HEAD, revision: eaf81808d5aa19ee4b9b8980ac5feeaec51408ac)
  build user:       root@e8effea18820
  build date:       20190814-17:42:12
  go version:       go1.12.8
```
* Prometheus configuration file:

```
global:
  scrape_interval: 1m
  scrape_timeout: 10s

scrape_configs:
  - job_name: 'prometheus'
    metrics_path: '/prometheus/metrics'
    static_configs:
      - targets: ['prometheus:9090']
        labels:
          group: 'production'

  - job_name: 'speedtest'
    metrics_path: '/probe'
    scrape_interval: 10m
    scrape_timeout: 2m
    static_configs:
      - targets: ['prometheus_speedtest:9516']
        labels:
          group: 'production'
    relabel_configs:
      - source_labels: ['__address__']
        target_label: 'instance'
        regex: '^(.*):[0-9]+$'

  - job_name: 'piaware_prober'
    metrics_path: '/probe'
    params:
      module: ['http_2xx']
    static_configs:
      - targets: ['http://192.168.1.123:80']
        labels:
          group: 'production'
    relabel_configs:
      - source_labels: ['__address__']
        target_label: '__param_target'
      - target_label: 'instance'
        replacement: 'piaware'
      - target_label: '__address__'
        replacement: 'prometheus_blackbox:9115'
      - source_labels: ['__param_module']
        target_label: 'module'

rule_files:
  - 'rules/*.yml'
```

* Docker configuration

```
  prometheus:                                                                      
    image: 'prom/prometheus:v2.12.0-rc.0'                                           
    user: '127:137'                                                                
    networks:                                                                      
      - 'prometheus_network'                                                                                                      
    volumes:                                                                       
      - type: 'bind'                                                               
        source: './monitoring/prometheus/prometheus.yml'                           
        target: '/prometheus/prometheus.yml'                                       
        read_only: true                                                            
      - type: 'bind'                                                               
        source: './monitoring/prometheus/rules'                                    
        target: '/prometheus/rules'                                                
        read_only: true                                                            
      - type: 'bind'                                                               
        source: '/var/prometheus/data'                                             
        target: '/prometheus/data'                                                 
    command:                                                                       
      - '--query.lookback-delta=25m'                                               
      - '--storage.tsdb.path=/prometheus/data'                                     
      - '--storage.tsdb.retention=10y'                                             
      - '--web.external-url=http://prometheus:9090/prometheus/'                    
    restart: 'unless-stopped'
```

**Comment by brian-brazil** [comment]
What does the active queries file look like?

**Comment by jeanralphaviles** [comment]
Looks to be a 20k file full of nothing.

```bash
/prometheus/data $ ls -lh queries.active 
-rw-r--r--    1 127      137        19.5K Sep 22 15:20 queries.active
/prometheus/data $ xxd queries.active | head
00000000: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00000010: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00000020: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00000030: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00000040: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00000050: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00000060: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00000070: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00000080: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00000090: 0000 0000 0000 0000 0000 0000 0000 0000  ................
/prometheus/data $ xxd queries.active | tail
00004d90: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00004da0: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00004db0: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00004dc0: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00004dd0: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00004de0: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00004df0: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00004e00: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00004e10: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00004e20: 00 
```

**Comment by jeanralphaviles** [comment]
If I rename `queries.active` to `queries.active.old` Prometheus starts and runs as expected.

Diffing the two files I do see that the new `queries.active` file is very slightly different.

```bash
/prometheus/data $ xxd queries.active > /tmp/active
/prometheus/data $ xxd queries.active.old > /tmp/active.old
/prometheus/data $ diff /tmp/active.old /tmp/active
--- /tmp/active.old
+++ /tmp/active
@@ -1,4 +1,4 @@
-00000000: 0000 0000 0000 0000 0000 0000 0000 0000  ................
+00000000: 5b00 0000 0000 0000 0000 0000 0000 0000  [...............
 00000010: 0000 0000 0000 0000 0000 0000 0000 0000  ................
 00000020: 0000 0000 0000 0000 0000 0000 0000 0000  ................
 00000030: 0000 0000 0000 0000 0000 0000 0000 0000  ................
```

**Comment by roidelapluie** [comment]
This was fixed in 2.13.1. This issue was not close because it was merged in the release branch.
